### PR TITLE
feat(elements|ino-currency-input): add component

### LINKED ISSUE #476: Currency Input
### Feature Request
The currency feature is currently directly integrated to the ino-input. This leads to a couple of problems in formatting the currency and increases the complexity.

### Description of the proposed solution
The currency input should be a separate component composing around an input field to only provide these features.

```
ino-currency-input
|  ino-input
```

### Additional context

Component already exists and must be migrated.

### PR DESCRIPTION
**Pull request by pfecht**
Closes #476 

This PR contains:
* a new component ino-currency-input which invents a composition concept (with `UserInputInterceptor`) to reduces duplication in "special" input elements.
* refactors to the `ino-input` to be grouped by properties, events, methods, lifecyles, ...
* refactors the `ino-input` to use a `<label>` instead of `<div>` wrt. to the documentation of mdc.

## Proposed Changes

- ...

## Things to check

- [ ] Does the change need to be documented?
- [ ] Does any existing example code needs to be updated?
- [ ] Is the change properly tested?
- [ ] Is it helpful to provide another example to demonstrate the new feature?
- [ ] Are there other code lines that need to be modified?

### CODE REVIEW

File: packages/elements/src/components/ino-currency-input/ino-currency-input.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+  connectedCallback() {
+    // This is needed if the currency input is reattached to the dom (for instance in a popover)
+    this.initialize();
+
+    // Setup listener manually to only handle child events and catch non-bubbeling focus / blur.
+    // @Listen would also trigger on valueChange of currency-input.
+    this.inoInputEl?.addEventListener('inoFocus', this.handleFocus);
+    this.inoInputEl?.addEventListener('inoBlur', this.handleBlur);
+    this.inoInputEl?.addEventListener('valueChange', this.handleValueChange);
```
Comment (BenPag) [inline packages/elements/src/components/ino-currency-input/ino-currency-input.tsx:80]: Duplicate code fragment here

Comment (pfecht) [reply packages/elements/src/components/ino-currency-input/ino-currency-input.tsx]: Thanks, removed it :)

File: packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  min: '',
+  max: '',
+  outlined: true,
+  name: '',
+  placeholder: '',
+  readonly: false,
+  required: false,
+  showLabelHint: false,
+  unit: '',
+  value: '10.50',
```
Comment (janivo) [inline packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts:80]: The knob element seems to be not quite right yet.
image

File: packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  currencyLocale: '',
+};
+Playground.argTypes = {
+  currencyLocale: {
+    control: 'select',
+    options: ['', 'de-DE', 'en-US'],
```
Comment (janivo) [inline packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts:86]: This seems to have no effect:
image

**Review by janivo — CHANGES_REQUESTED** [review]

File: packages/elements/src/components/ino-currency-input/ino-currency-input.tsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+  // -----
+  // Currency logic
+
+  private getSeperator() {
+    return this.toCurrencyString(0).charAt(1);
+  }
+
+  private toCurrencyString(value: string | number): string {
+    if (value || value === 0) {
```
[the remark this answers is not in the document]
Comment (pfecht) [reply packages/elements/src/components/ino-currency-input/ino-currency-input.tsx:160]: value is a string which means `!"0" == true`. Bit weird here, added a comment.

File: packages/elements/src/components/ino-currency-input/ino-currency-input.tsx [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    this.inoInputEl?.addEventListener('inoBlur', this.handleBlur);
+    this.inoInputEl?.addEventListener('valueChange', this.handleValueChange);
+  }
+
+  // -------
+  // Event Handler
+
+  private handleValueChange = (e: CustomEvent) => {
+    let value = e.detail;
+    value = !value ? null : this.fromCurrencyString(value);
```
[the remark this answers is not in the document]
Comment (pfecht) [reply packages/elements/src/components/ino-currency-input/ino-currency-input.tsx:101]: value is a string which means !"0" == true. Bit weird here, added a comment.

File: packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts (hunk 2 above)
Comment (pfecht) [reply packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts:80]: Seems to be a problem for storybook if the type is `string | number`. I hardcoded the argtype, now its  working :)

File: packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts (hunk 3 above)
Comment (pfecht) [reply packages/storybook/src/stories/ino-currency-input/ino-currency-input.stories.ts:86]: For me its working (changing the thousands separator and decimal separator)

File: yarn.lock [hunk 6]
```diff
@@ -11698,7 +11698,7 @@ cuint@^0.2.2:
 
 currency.js@^1.2.2:
   version "1.2.2"
-  resolved "https://registry.npmjs.org/currency.js/-/currency.js-1.2.2.tgz#ae84daf68798051097ad3feb2d022ad08994f526"
+  resolved "https://registry.yarnpkg.com/currency.js/-/currency.js-1.2.2.tgz#ae84daf68798051097ad3feb2d022ad08994f526"
```
Comment (BenPag) [inline yarn.lock:11701]: removed in `package.json` but still exists in `yarn.lock`. Is that correct?

**Comment by janivo** [comment]
Could you resolve the conflicts and update the `yarn.lock` as @BenPag pointed out @pfecht? Everything else seems fine

**Review by janivo — APPROVED** [review]

**Pull request merged**